### PR TITLE
CI: fix installer deploy jobs

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR introduces a specific template for installer deploy jobs, as we don't want to upload them into `dd-release-artifacts/datadog-agent`, but in `dd-release-artifacts/datadog-installer` instead.
The release management tooling expects the artifacts to live in a folder named as the product its deploying, meaning our installer jobs were broken since we split the installer as a dedicated product.

### Motivation

Fixing our non-OCI deployment jobs

### CODE REVIEW

File: .gitlab/deploy_packages/nix.yml [hunk 1]
```diff
@@ -219,26 +219,61 @@ deploy_staging_iot_agent:
...
+  script:
+    - $S3_CP_CMD --recursive --exclude "*" --include "*_7.*${PACKAGE_ARCH}.deb" "$OMNIBUS_PACKAGE_DIR" "s3://dd-release-artifacts/datadog-installer/$CI_PIPELINE_ID/deb/${PACKAGE_ARCH}/"
+
+.deploy_installer_rpm:
+  rules:
+    !reference [.on_deploy]
+  resource_group: rpm_bucket
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/gitlab_agent_deploy:$DATADOG_AGENT_BUILDERS
+  stage: deploy_packages
+  tags: ["arch:amd64"]
```
Comment (raphaelgavache) [inline .gitlab/deploy_packages/nix.yml:247]: shouldn't the arch tag be overwritten below?

Comment (chouquette) [reply .gitlab/deploy_packages/nix.yml:247]: This is the runner arch, not the package one. We *could* run these jobs on arm (maybe we should actually) but this is orthogonal to the arch of the package which is being deployed

**Review by raphaelgavache — APPROVED** [review]

File: .gitlab/deploy_packages/nix.yml [hunk 2]
```diff
@@ -219,26 +219,61 @@ deploy_staging_iot_agent:
...
 
 # Datadog Installer
+
+# Common jobs
+
+# The installer is a special case because it's built by the datadog-agent pipeline, but
+# is expected to be deployed in a dedicated folder, so we need to define a special base which
+# fetches packages from a custom value of S3_RELEASE_ARTIFACTS_URI as the default value
+# include datadog-agent as a product
+.deploy_installer_deb:
```
Comment (alopezz) [inline .gitlab/deploy_packages/nix.yml:229]: Any reason not to move these templates to the deploy_common.yml file for consistency? Not a big deal, though.

File: .gitlab/deploy_packages/nix.yml [hunk 3]
```diff
@@ -219,26 +219,61 @@ deploy_staging_iot_agent:
...
+  rules:
+    !reference [.on_deploy]
+  resource_group: deb_bucket
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/gitlab_agent_deploy:$DATADOG_AGENT_BUILDERS
+  stage: deploy_packages
+  tags: ["arch:amd64"]
+  before_script:
+    - ls $OMNIBUS_PACKAGE_DIR
+  script:
+    - $S3_CP_CMD --recursive --exclude "*" --include "*_7.*${PACKAGE_ARCH}.deb" "$OMNIBUS_PACKAGE_DIR" "s3://dd-release-artifacts/datadog-installer/$CI_PIPELINE_ID/deb/${PACKAGE_ARCH}/"
```
Comment (alopezz) [inline .gitlab/deploy_packages/nix.yml:239]: Might be worth adding something like
```yaml
S3_RELEASE_INSTALLER_ARTIFACTS_URI: s3://dd-release-artifacts/datadog-installer
```
right after https://github.com/DataDog/datadog-agent/blob/92d6c0a59fb1dfe48f96b9d66ce3aa67f85aa612/.gitlab-ci.yml#L147

Just to keep those centralised.

**Review by alopezz — APPROVED** [review]

File: .gitlab/deploy_packages/nix.yml (hunk 2 above)
Comment (chouquette) [reply .gitlab/deploy_packages/nix.yml:229]: I hesitated but I guess it makes more sense, I'll update

File: .gitlab/deploy_packages/nix.yml (hunk 3 above)
Comment (chouquette) [reply .gitlab/deploy_packages/nix.yml:239]: Good point!

**Review by KSerrania — APPROVED** [review]

**Pull request merged**